server/src/main/java/org/elasticsearch/action/admin/indices/rollover/RolloverConfiguration.java

```diff
@@ -0,0 +1,434 @@
+        PARSER.declareString((valueParser, s) -> valueParser.addMinIndexSizeCondition(s, MinSizeCondition.NAME), MIN_SIZE_FIELD);
+        PARSER.declareString(
+            (valueParser, s) -> valueParser.addMinPrimaryShardSizeCondition(s, MinPrimaryShardSizeCondition.NAME),
+            MIN_PRIMARY_SHARD_SIZE_FIELD
+        );
+        PARSER.declareString(
+            (valueParser, s) -> valueParser.addMinPrimaryShardDocsCondition(s, MinPrimaryShardDocsCondition.NAME),
+            MIN_PRIMARY_SHARD_DOCS_FIELD
+        );
+    }
+
+    // The conditions that have concrete values
+    private final RolloverConditions concreteConditions;
+    // The conditions that have the value `auto`, currently only max_age is supported
+    private final Set<String> automaticConditions;
+
+    public RolloverConfiguration(RolloverConditions concreteConditions, Set<String> automaticConditions) {
+        this.concreteConditions = concreteConditions;
+        this.automaticConditions = automaticConditions;
+    }
+
+    public RolloverConfiguration(RolloverConditions concreteConditions) {
+        this.concreteConditions = concreteConditions;
+        this.automaticConditions = Set.of();
+    }
+
+    public RolloverConfiguration(StreamInput in) throws IOException {
+        concreteConditions = new RolloverConditions(in);
+        automaticConditions = in.readSet(StreamInput::readString);
+    }
+
+    public static RolloverConfiguration fromXContent(XContentParser parser) throws IOException {
+        ValueParser valueParser = new ValueParser();
+        PARSER.parse(parser, valueParser, null);
+        return valueParser.getRolloverConfiguration();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeWriteable(concreteConditions);
+        out.writeCollection(automaticConditions, StreamOutput::writeString);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        for (Condition<?> condition : concreteConditions.getConditions().values()) {
+            builder.field(condition.name, condition.valueToHumanReadableString());
+        }
+        for (String automaticCondition : automaticConditions) {
+            builder.field(automaticCondition, "auto");
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    /**
+     * This version of the toXContent method takes into consideration the retention and creates an XContent object
+     * where the automatic values are evaluated and have the label [automatic]
+     */
+    public XContentBuilder toXContent(XContentBuilder builder, Params ignored, TimeValue retention) throws IOException {
+        builder.startObject();
+        for (Condition<?> condition : concreteConditions.getConditions().values()) {
+            builder.field(condition.name, condition.valueToHumanReadableString());
```

Comment:
We cannot use this because this serializes the `Condition<Long>` to a long, so when we are parsing it we can only parse it as a long. So we cannot give the meaningful error in case it's `auto` and we treat the parsing from a setting and parsing from `XContent` differently.

At least this why I made this choice, what are your thoughts about this now that you know the context?

Reply:
After an offline discussion we decided to remove the XContent parser all together, this makes this change not necessary.